### ISSUE TITLE
How to Get Custom Embedding Size from Tabnet

### ISSUE THREAD

**Issue by Hazqeel09** [dreamquark-ai/tabnet#445]
I am trying to do something similar with https://github.com/dreamquark-ai/tabnet/issues/167 where I want to use TabNet as feature extractor. But I noticed that the output dimensions from https://github.com/dreamquark-ai/tabnet/issues/167 is in the size of how many classes for predictions.

I want to duplicate the method of this paper (https://ieeexplore.ieee.org/document/9658729) where TabNet produced embedding of 64 dimensions.

![image](https://user-images.githubusercontent.com/33782405/207043039-d02238a7-3a9a-4a16-b185-80c283f33966.png)

**Comment by Hazqeel09** [comment]
I'm sorry, I'm not quite familiar on how to use that. Been trying for weeks.

Does it mean I can use TabNetEncoder like a neural network layer? So I can combine it into a big layer like the paper?

**Comment by Optimox** [comment]
Yes you can use this as part of a bigger model. Any `torch.nn.Module` can be used as build block.

So you'll need your part of your inputs going to the bert model, the other going the TabNetEncoder and the you contenate the outputs to pass them to the 2D convolutional layer.
